### PR TITLE
using environment variables to set up server port

### PR DESCRIPTION
**Pull request by AadilLatif**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -29,6 +29,7 @@ dependencies = [
   "psutil",
   "click",
   "pyyaml~=6.0",
+  "kubernetes",
```
Comment (josephmckinsey) [inline pyproject.toml:32]: I think we need a >=29 on this to get the CI tests to pass.

P.S. We can probably remove a bunch of the test dependencies related to formatting and all that right now.

File: src/oedisi/tools/cli_tools.py [hunk 2]
```diff
@@ -5,7 +5,8 @@
 import json
 import os
 from pathlib import Path
-
+from kubernetes import client, config
```
Comment (josephmckinsey) [inline src/oedisi/tools/cli_tools.py:8]: config is an unused import

File: src/oedisi/tools/cli_tools.py [hunk 3]
```diff
@@ -156,57 +156,97 @@ def validate_optional_inputs(
...
+    return clean_model
+
 def create_kubernetes_deployment(
     wiring_diagram: WiringDiagram, target_directory, broker_port, node_port
 ):
     kube_folder = os.path.join(target_directory, "kubernetes")
     if not os.path.exists(kube_folder):
         os.mkdir(kube_folder)
+    
+    SERVICE_NAME = "oedisi-service"
```
Comment (josephmckinsey) [inline src/oedisi/tools/cli_tools.py:192]: Make into global constant instead. Maybe in types.common?

File: src/oedisi/tools/cli_tools.py [hunk 4]
```diff
@@ -156,57 +156,97 @@ def validate_optional_inputs(
...
+            kind="Pod",
+            metadata=client.V1ObjectMeta(
+                name=f"{fixed_container_name}-pod",
+                labels ={"app" : APP_NAME},
+            ),
+            spec=client.V1PodSpec(
+                containers=[my_container],
+                hostname=component.name.replace("_", "-"),
+                subdomain="oedisi-service",
+                #node_port=int(component.container_port), #maybe
```
Comment (josephmckinsey) [inline src/oedisi/tools/cli_tools.py:243]: We probably want to remove this comment or adjust it.

File: src/oedisi/tools/cli_tools.py [hunk 5]
```diff
@@ -156,57 +156,97 @@ def validate_optional_inputs(
...
+        else:
+            key_name = k    
+                
+        if isinstance(v, dict):
+            clean_model[key_name] = drop_null_values(v)
+        elif isinstance(v, list):
+            new_list = []
+            for val in v:
+                new_list.append(drop_null_values(val))
+            clean_model[key_name] =new_list
```
Comment (josephmckinsey) [inline src/oedisi/tools/cli_tools.py:180]: We may want to run ruff to format this to get rid of things like  `=new_list`.

File: src/oedisi/types/common.py [hunk 6]
```diff
@@ -10,6 +10,7 @@
 class DefaultFileNames(str, Enum):
     INPUT_MAPPING = "input_mapping.json"
     STATIC_INPUTS = "static_inputs.json"
+    WIRING_DIAGRAM = "system.json"
```
Comment (josephmckinsey) [inline src/oedisi/types/common.py:13]: What's the use of this? It seems to be completely internal to the broker?

**Review by josephmckinsey — CHANGES_REQUESTED** [review]
It seems that the "host" parameter is completely unused in our system.json? We should probably eliminate that.

File: src/oedisi/tools/cli_tools.py (hunk 2 above)
Comment (AadilLatif) [reply src/oedisi/tools/cli_tools.py:8]: removed

File: src/oedisi/tools/cli_tools.py (hunk 3 above)
Comment (AadilLatif) [reply src/oedisi/tools/cli_tools.py:192]: replaceed with KUBERNETES_SERVICE_NAME. moved to common

File: src/oedisi/tools/cli_tools.py (hunk 4 above)
Comment (AadilLatif) [reply src/oedisi/tools/cli_tools.py:243]: removed

File: src/oedisi/tools/cli_tools.py (hunk 5 above)
Comment (AadilLatif) [reply src/oedisi/tools/cli_tools.py:180]: ruff weirdly did not catch it 

src\oedisi\tools\cli_tools.py:161:5: D103 Missing docstring in public function
src\oedisi\tools\cli_tools.py:186:5: D103 Missing docstring in public function
src\oedisi\tools\cli_tools.py:208:41: F541 [*] f-string without any placeholders
src\oedisi\tools\cli_tools.py:251:5: D103 Missing docstring in public function
src\oedisi\tools\cli_tools.py:260:5: D103 Missing docstring in public function

fixed it

File: src/oedisi/types/common.py (hunk 6 above)
Comment (AadilLatif) [reply src/oedisi/types/common.py:13]: DefaultFileNames.WIRING_DIAGRAM is used in the sigdal example. ~\sgidal-example\broker\server.py line 209.
I want to be consistent and not use hard-coded file names

File: pyproject.toml (hunk 1 above)
Comment (AadilLatif) [reply pyproject.toml:32]: done

File: src/oedisi/types/common.py (hunk 6 above)
Comment (AadilLatif) [reply src/oedisi/types/common.py]: moved it after reading comment on the example repo

**Review by josephmckinsey — APPROVED** [review]
I got an unusual untracked file when running tests locally:

	tests/unit_tests/test_oedisi_tools/broker/docker-compose.yml
	tests/unit_tests/test_oedisi_tools/component1/Dockerfile
	tests/unit_tests/test_oedisi_tools/component2/Dockerfile
	
I expect the later two, but I was more surprised at the docker-compose.yml file. Do you know why that's there?

Also, right now this will only work when all components are in the top-level directory and we are building to a folder one level down. We'll have to either remember to document this, check that assumption in the cli, or fix it by using the component definitions directly.

**Review by josephmckinsey — CHANGES_REQUESTED** [review]
I just realized by testing that the new `-m` method doesn't create the build folder since that was done in the initialize_federates

**Comment by AadilLatif** [comment]
> I just realized by testing that the new `-m` method doesn't create the build folder since that was done in the initialize_federates

fixed

**Comment by AadilLatif** [comment]
> I got an unusual untracked file when running tests locally:
> 
> ```
> tests/unit_tests/test_oedisi_tools/broker/docker-compose.yml
> tests/unit_tests/test_oedisi_tools/component1/Dockerfile
> tests/unit_tests/test_oedisi_tools/component2/Dockerfile
> ```
> 
> I expect the later two, but I was more surprised at the docker-compose.yml file. Do you know why that's there?
> 
> Also, right now this will only work when all components are in the top-level directory and we are building to a folder one level down. We'll have to either remember to document this, check that assumption in the cli, or fix it by using the component definitions directly.

Broker uses docker-compose.yml ro figure out ports for all participating containers. It is needed, which is why I make a copy

For the second comment, I will make a note

**Comment by AadilLatif** [comment]
> It seems that the "host" parameter is completely unused in our system.json? We should probably eliminate that.

Yes it can be removed now. Iwas using it for I stuff we no longer use

**Review by josephmckinsey — APPROVED** [review]
We'll can remove the feeder from the BrokerConfig when we deal with the sensor issue.

**Pull request merged**